pkg/operator/operator.go

```diff
@@ -94,13 +94,14 @@ func NewOperator() (context.Context, *Operator) {
 	kubernetesInterface := kubernetes.NewForConfigOrDie(config)
 	configMapWatcher := informer.NewInformedWatcher(kubernetesInterface, system.Namespace())
 
-	// Settings
-	settingsStore := settingsstore.WatchSettingsOrDie(ctx, kubernetesInterface, configMapWatcher, apis.Settings.List()...)
-
 	// Logging
 	logger := NewLogger(ctx, component, config, configMapWatcher)
 	ctx = logging.WithLogger(ctx, logger)
 
+	// Settings
+	settingsStore := settingsstore.NewWatcherOrDie(ctx, kubernetesInterface, configMapWatcher, apis.Settings.List()...)
+	settingsStore.WaitForConfigMapsOrDie(ctx)
```

Comment:
Are there any cases where we don't want to wait as part of the NewWatcherOrDie function?